tests/ruby/misc_spec.rb

```diff
@@ -294,6 +294,30 @@
 
         expect(processes.primary.count_query("DISCARD ALL")).to eq(10)
       end
+
+      it "Respects tracked parameters on startup" do
+        conn = PG::connect(processes.pgcat.connection_string("sharded_db", "sharding_user", parameters: { "application_name" => "my_pgcat_test" }))
+
+        expect(conn.async_exec("SHOW application_name")[0]["application_name"]).to eq("my_pgcat_test")
+        conn.close
+      end
+
+      it "Respect tracked parameter on set statemet" do
+        conn = PG::connect(processes.pgcat.connection_string("sharded_db", "sharding_user"))
+
+        conn.async_exec("SET application_name to 'my_pgcat_test'")
+        expect(conn.async_exec("SHOW application_name")[0]["application_name"]).to eq("my_pgcat_test")
+      end
+
+
+      it "Ignore untracked parameter on set statemet" do
+        conn = PG::connect(processes.pgcat.connection_string("sharded_db", "sharding_user"))
+        orignal_statement_timeout = conn.async_exec("SHOW statement_timeout")[0]["statement_timeout"]
+
+        conn.async_exec("SET statement_timeout to 1500")
```

Comment:
I'm not sure that this behavior is desired. We are actually now ignoring commands that the pooler and database would normally accept.

Reply:
This is actually equivalent to what we had before, since we send a discard all after every non-transaction statement in transaction mode. The point of these tests is that across clients "session" in transaction mode, the tracked parameters are preserved (like pgbouncer) and untracked ones are not.

Reply:
That behavior is optional and controlled by a config. Is this still the case?

Reply:
Yes it's still controlled by that config